src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,213 @@ int WaterTracksObj::Update(int ms_elapsed)
+                    g_theWaterTracksRenderSystem->Un_Bind_Track(track);
+
+                    if (track2 != nullptr) {
+                        g_theWaterTracksRenderSystem->Un_Bind_Track(track2);
+                    }
+
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                }
+            } else {
+                delete_track_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_INSERT) & 0x8001) != 0) {
+                if (change_type_reset == true) {
+                    change_type_reset = false;
+                    current_wave_type++;
+
+                    if (current_wave_type > WAVE_TYPE_RADIAL) {
+                        current_wave_type = WAVE_TYPE_POND;
+                    }
+
+                    Utf16String str;
+                    str.Format(L"Wave Type: %hs", s_waveTypeInfo[current_wave_type].wave_name);
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                change_type_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_F7) & 0x8001) != 0) {
+                if (save_tracks_reset == true) {
+                    g_theWaterTracksRenderSystem->Save_Tracks();
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                    Utf16String str;
+                    str.Format(L"Saved Tracks");
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                save_tracks_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_F8) & 0x8001) != 0) {
+                if (load_tracks_reset == true) {
+                    load_tracks_reset = false;
+                    g_theWaterTracksRenderSystem->Reset();
+                    g_theWaterTracksRenderSystem->Load_Tracks();
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                    Utf16String str;
+                    str.Format(L"Loaded Tracks");
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                save_tracks_reset = true;
+            }
+        }
+
+        if (have_start == true && have_end == true) {
+            ICoord2D coord;
+            coord.x = point.x;
+            coord.y = point.y;
+            g_theTacticalView->Screen_To_Terrain(&coord, &terrain_point_end);
+            int x = terrain_point_end.x - terrain_point_start.x;
+            int y = terrain_point_end.y - terrain_point_start.y;
+
+            if (s_waveTypeInfo[current_wave_type].final_width >= GameMath::Sqrt(x * x + y * y)) {
+                g_theDisplay->Draw_Line(mouse_anchor.x, mouse_anchor.y, point.x, point.y, 1.0f, 0xFFCCCCFF);

```

Comment:
Do we not have some sort of macro or constexpr function for writing colours. It would be much clearer if this was `COLOR(255, 204, 204, 255)` and a comment next to it as to what name that color is.